File: src/main/java/phoenix/init/PhoenixBiomes.java
package phoenix.init;

import net.minecraft.world.biome.Biome;
import net.minecraftforge.fml.RegistryObject;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
import net.minecraftforge.registries.DeferredRegister;
import net.minecraftforge.registries.ForgeRegistries;
import phoenix.Phoenix;
import phoenix.world.biomes.HeartVoidBiome;
import phoenix.world.biomes.UnderBiome;

public class PhoenixBiomes
{
    public static final DeferredRegister<Biome> BIOMES = new DeferredRegister(ForgeRegistries.BIOMES, Phoenix.MOD_ID);

    public static final RegistryObject<Biome> UNDER     = BIOMES.register("under", UnderBiome::new);
    public static final RegistryObject<Biome> HEARTVOID = BIOMES.register("heart_void", HeartVoidBiome::new);

    public static void register()
    {
        BIOMES.register(FMLJavaModLoadingContext.get().getModEventBus());
    }
}

File: src/main/kotlin/phoenix/init/PhoenixBiomes.kt
package phoenix.init

import net.minecraft.world.biome.Biome
import net.minecraftforge.registries.ForgeRegistries
import phoenix.Phoenix
import phoenix.utils.GenerationUtils
import phoenix.world.biomes.UnderBiome
import phoenix.world.builders.Builders
import thedarkcolour.kotlinforforge.forge.KDeferredRegister
import thedarkcolour.kotlinforforge.forge.MOD_BUS

object PhoenixBiomes
{
    val BIOMES = KDeferredRegister(ForgeRegistries.BIOMES, Phoenix.MOD_ID)

    val UNDER     by BIOMES.register("under", ::UnderBiome)
    val HEARTVOID by BIOMES.register("heart_void", ::HeartVoidBiome)

    fun register() =  BIOMES.register(MOD_BUS)
}

class HeartVoidBiome : Biome(GenerationUtils.defaultSettingsForEnd(Builders.HEARTVOID, Builders.HEARTVOID_CONFIG))